cache/FIFO_FH.h

```diff
@@ -959,11 +974,7 @@ inline void FIFO_FHCache<TKey, TValue, THash>::melt_chunk() {
     chunkNodeHead->m_prev = &m_head;
     lastNode->m_next = nextNode;
     nextNode->m_prev = lastNode;
-
-    while(chunkNodeHead != lastNode) {
-      m_fasthash->remove_key(chunkNodeHead->m_key);
-      chunkNodeHead = chunkNodeHead->m_next;
-    }
+    globalCounter += 1;
```

Comment:
nit: globalCounter already atomic, so unlocking the lock explicitly one line earlier should be just slightly more efficient than relying on the end of the scope to unlock